index.bs

```diff
@@ -3168,7 +3173,7 @@ To <dfn>get the navigation info</dfn>, given |navigable| and |navigation status|
 #### The browsingContext.ReadinessState Type #### {#type-browsingContext-ReadinessState}
 
 <pre class="cddl remote-cddl">
-browsingContext.ReadinessState = "none" / "interactive" / "complete"
+browsingContext.ReadinessState = "none" / "committed" / "interactive" / "complete"
```

Comment:
So this means that we are fine for the `wait` argument of `browsingContext.navigate` to allow `committed` as well? Asking because it causes a side-effect.

Reply:
Could you please clarify what you mean by the side-effect?

Reply:
So far we spoke about a committed event but not that we have another readiness state that could be used by other APIs to control the navigation flow like for `browsingContext.navigate`. In the current form this value for the `wait` argument wouldn't be denied but also not used in [await a navigation](https://w3c.github.io/webdriver-bidi/#await-a-navigation) because we don't have a check. So it's basically a no-op.

Reply:
I believe the value is checked on line 3045 (part of the await navigation algorithm you linked) so it does not appear to be no-op. I think it's needed for internal consistency but we can discuss it separately.

Reply:
Ah indeed. Even through I opened the preview in a different tab I actually checked the public BiDi spec page. Sorry for that.

While investigating some more Playwright tests I saw the following:

> await page.goto(server.EMPTY_PAGE, { waitUntil: 'commit' })

Is that a condition that would help Puppeteer users as well that we should add to the navigate steps to return on the committed step? Or do it in a follow-up PR?

Reply:
Yeah, in Puppeteer we always waited for commit by default and switched to waiting for interactive by default for tests to pass and handle errors but we would prefer to roll back to waiting for commit as it could be a little more robust in some situation. I think we do not expose it as an option on the API though but maybe we should.

> Is that a condition that would help Puppeteer users as well that we should add to the navigate steps to return on the committed step? Or do it in a follow-up PR?

I didn't get the question, sorry. I believe this PR already returns navigate steps on "committed" if it is configured so. We could extract the whole waitUntil part into a separate PR if it makes it easier.

Reply:
No that is fine. So lets keep it in. Thanks.

It would be great if you could actually write some WPT tests for the event and the wait argument given that it seems to be easy for you to add support for `committed` (CDP already supports it).

Reply:
I filed https://github.com/GoogleChromeLabs/chromium-bidi/issues/2982 to write the tests as part of the implementation.

Reply:
I tend to think we want this to continue to match the `document.readyState` property, but for the purposes of awaiting a navigation make "none" work like "committed", unless we have a use case where you really don't even care if the navigation started or not, but still need the return value of the command for some reason.

Reply:
I guess if we are not concerned about breaking changes, I am in favor of replacing `none` to mean `committed`  (cc @sadym-chromium)

Reply:
> I am in favor of replacing `none` to mean `committed`

YES!

Reply:
Updated the PR, please take a look.

Reply:
That's actually a great idea. I support that as well because it will reduce failures due to process switches (not in all cases like redirects but most of them). We potentially might break some users but lets see how severe this will be. Benefit for sure will be that we will know the target URL at that point and clearly can return a proper navigation info or error response.